### ISSUE TITLE
Just some things to share about OBS audio status

### ISSUE THREAD

**Issue by x0rx** [pkviet/obs-studio#43]
Well, I noticed there's a growing interest in some audio features not present yet in main OBS, maybe because there's more users using OBS to stream video-chats. Recently I see this petition in OBS ideas forum: https://ideas.obsproject.com/posts/965/additional-aux-send-monitor-channel-or-virtual-audio-output

And this is my comment:

```
it would be a very nice feature to have:

-    A main monitor output (already done, for the streamer, obs user)
-    An aux monitor output (as Virtual cam does, to return a custom sound mix to chat guests, ...)
-    A Master output volume fader (main stream/record). With filters, lufs metering (loudness) to simply put compressor/limiter here and not one by one in each input mixer tracks.
-    Two monitor output volume faders (main monitor, aux monitor).
-    And a redesign of the Advanced Audio Properties monitor section, with a matrix of check boxes, same as the multitrack audio output OR adding and option to do this sending to main/monitor/aux monitor per track, from the input mixer panel, without leaving the main OBS screen/opening another window.

Some of this is already done in OBS Music Edition and would be great if this features can be enabled in OBS Studio.
```

I don´t know the work this changes involve nor the current interest in the main OBS developer community about this audio issues.
Once the work is done for a monitor output maybe is not much work, to do a second one. I don't know.

When I knew about Music Edition, reading posts in the forum, it seems that the work done in this fork as master mixer, etc.. is not a thing that deserves much interest in order to be added to the main OBS. How is the current situation in the developer community relating to this things? Can we expect some audio work done here to be added to main OBS?

So, while this things are considered, first i want to thank pkviet and others for the time and work done, and if you like and have the time, maybe you can consider adding this features to ME in future releases.

I know that there's a lot of use cases and users with specific needs, some need easy things, some need something like a DAW, but I would consider this features as basic ones at this time, and this would solve a lot of problems people have with audio, without the need of third party software as Voicemeeter, etc.. that for most users is like entering in another space of understanding. How Windows audio works, apps, drivers, devices, etc... is by itself a day to day enough problem for many users.

Have a good day everyone in this crazy times we are living.

**Comment by Andersama** [comment]
I'm definitely slightly confused for what you're asking for. You're right this fork was specifically made as an edit to add a master mixer / master mixer stage. It sounds like you're asking for a second device to monitor audio from, and a master fader (ME's definitely got that).

If I'm not completely lost I suspect a combination of this fork and https://obsproject.com/forum/resources/audio-monitor.1186/
would help.

Between me and pkv, I don't know that pkv would revisit this to edit the advanced audio properties section or a second monitoring device, I'm definitely not (sorry). The advanced audio properties popup has always been a pain point because in part for regular users it's not exactly made clear how the backend of OBS works so it's not necessarily obvious for example what the "tracks" 1,2,3,4,5,6 are and how they're used.

Part of the reason for the master mixer was to deal with that bit of the audio processing pipeline which was inaccessible to the user. In the process we also made it a point to allow people to add filters to that last stage of the pipeline to do proper global audio processing.

> Can we expect some audio work done here to be added to main OBS?

To be frank, no. The reason I no longer other doing anything w/ obs is that Jim has extreme issues with me and Jim's made an ultimatum that he does not want any code to appear in obs with my name on it. If you're making noise about it in the discord or anywhere else the unfortunate reality is and that's it's probably not going to go anywhere, Jim's expressly aware I contributed to this (we pitched this to him ages ago). TDLR; It's highly unlikely any of these features will appear in obs much less any other features I was working on at the time.

When I was on the dev team I tried to make a point that obs's feature set should be closer to that of a daw and other professional compositing software that's why this fork exists as it does. It's a culmination of several issues I had w/ obs's audio pipeline that needed a bit of love and so pkv and I and a few other people messed around with the internals until we had a version which worked for our needs and was closer to a daw as a whole. There were also, and I'd have no idea if it was ever patched several misleading aspects of obs's metering and other issues, fairly certain we fixed them in here. 

The driving force for a master mixer for example is that in obs unless you have one audio source mixed you may not necessarily be aware that your audio is clipping. EG: you may setup all your sources such that they appear to be ok per source, but the combined audio sent out to stream may in fact be clipping. It's not world ending obviously, but it's something for the professional. I'm pretty confident if I looked on the forums I can probably find a few people complaining about audio levels not matching what they've recorded...this would've made it more intuitive as to what is actually happening behind the scenes and would've made those people's lives less miserable.

There's an issue that you can't process your mixed audio as well, say for example if you wanted to make sure you never clipped (and although this wouldn't necessarily be good practice) you'd be tempted to have a hard limiter filter / plugin...but even if you wanted to that wasn't possible. There are obviously other pain points w/ audio in obs but it's kind of a mute point for me to even attempt to fix anything.

> I don´t know the work this changes involve nor the current interest in the main OBS developer community about this audio issues.

I'd have no idea what the "main" OBS developer community feels about this now, pkv'd have a better sense of that. There were only ever a few developers and I don't imagine there's too many new people since I left. Headcount wise there were at least two of us who thought this was important of the active people I could count on my hand. I can tell you that it wasn't particularly difficult to make this a thing. If you look at the commit history you're not going to be looking at much.

Glad it sounds like it's helped you a bit though.
